### PR TITLE
[notifications] Add permission hooks from modules factory

### PR DESCRIPTION
**Pull request by byCedric**
~~Blocked by #13860~~

# Why

Adds a permission hook from #13782 to `expo-notifications`. 

> Note, [just like `expo-location`](https://github.com/expo/expo/pull/13860), this requires generics in the permission hook factory to inherit the customized permission type.

# How

Exported permission hook and permission hook options type.

# Test Plan

See example in docblock.
# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-notifications/src/NotificationPermissionsModule.ts [hunk 1]
```diff
@@ -72,15 +72,36 @@ async function resolvePermissionAsync({
...
+ *
+ * @example
+ * ```ts
+ * const [status, requestPermission] = Notifications.usePermissions();
+ * ```
+ */
+const usePermissions = createPermissionHook({
+  getMethod: getPermissionsAsync,
+  // @ts-ignore - We need generics for this one
+  requestMethod: requestPermissionsAsync,
```
Comment (byCedric) [inline packages/expo-notifications/src/NotificationPermissionsModule.ts:98]: Todo(@byCedric): resolve this issue when generics are merged

File: packages/expo-notifications/CHANGELOG.md [hunk 2]
```diff
@@ -128,11 +129,9 @@ _This version does not introduce any user-facing changes._
 - Changed class responsible for handling Firebase events from `FirebaseMessagingService` to `.service.NotificationsService` on Android. ([#10558](https://github.com/expo/expo/pull/10558) by [@sjchmiela](https://github.com/sjchmiela))
 
   > Note that this change most probably will not affect you — it only affects projects that override `FirebaseMessagingService` to implement some custom handling logic.
-
```
Comment (cruzach) [inline packages/expo-notifications/CHANGELOG.md:131]: undo- does prettier do this auto for you? Mine doesn't seem to

File: packages/expo-notifications/CHANGELOG.md [hunk 3]
```diff
@@ -128,11 +129,9 @@ _This version does not introduce any user-facing changes._
...
 - Changed how you can override ways in which a notification is reinterpreted from a [`StatusBarNotification`](https://developer.android.com/reference/android/service/notification/StatusBarNotification) and in which a [`Notification`](https://developer.android.com/reference/android/app/Notification.html?hl=en) is built from defining an `expo.modules.notifications#NotificationsScoper` meta-data value in `AndroidManifest.xml` to implementing a `BroadcastReceiver` subclassing `NotificationsService` delegating those responsibilities to your custom `PresentationDelegate` instance. ([#10558](https://github.com/expo/expo/pull/10558) by [@sjchmiela](https://github.com/sjchmiela))
 
   > Note that this change most probably will not affect you — it only affects projects that override those methods to implement some custom handling logic.
-
```
Comment (cruzach) [inline packages/expo-notifications/CHANGELOG.md:135]: undo

**Review by cruzach — APPROVED** [review]
👍

File: packages/expo-notifications/CHANGELOG.md (hunk 2 above)
Comment (byCedric) [reply packages/expo-notifications/CHANGELOG.md:131]: I don't think I changed that, I just ran `et add-changelog ...`

**Pull request merged**